forge/test/mlir/llama/tests/test_specific_ops_llama32.py

```diff
@@ -0,0 +1,480 @@
+        def __init__(self):
+            super().__init__()
+
+        def forward(self, a, b):
+            return a + b
+
+    inputs = [torch.rand(shapes[0]), torch.rand(shapes[1])]
+
+    framework_model = Add()
+    fw_out = framework_model(*inputs)
+
+    compiled_model = forge.compile(framework_model, sample_inputs=inputs)
+    co_out = compiled_model(*inputs)
+
+    co_out = [co.to("cpu") for co in co_out]
+    fw_out = [fw_out] if isinstance(fw_out, torch.Tensor) else fw_out
+    assert all([compare_with_golden_pcc(golden=fo, calculated=co, pcc=0.99) for fo, co in zip(fw_out, co_out)])
+
+
+@pytest.mark.parametrize(
+    "inputs_and_dim",
+    [
+        ((1, 11, 32), (1, 11, 32), -1),
+        ((1, 32, 11, 32), (1, 32, 11, 32), -1),
+        ((1, 8, 11, 32), (1, 8, 11, 32), -1),
+    ],
+)
+def test_concat(inputs_and_dim):
+    in_shape1, in_shape2, dim = inputs_and_dim
+    print(in_shape1, in_shape2, dim)
+
+    class Concat(nn.Module):
+        def __init__(self):
+            super().__init__()
+
+        def forward(self, a, b):
+            return torch.cat((a, b), dim)
+
+    inputs = [torch.rand(in_shape1), torch.rand(in_shape2)]
+
+    framework_model = Concat()
+    fw_out = framework_model(*inputs)
+
+    compiled_model = forge.compile(framework_model, sample_inputs=inputs)
+    co_out = compiled_model(*inputs)
+
+    co_out = [co.to("cpu") for co in co_out]
+    assert compare_with_golden_pcc(golden=fw_out, calculated=co_out[0], pcc=0.99)
+
+
+@pytest.mark.parametrize("shapes", [(1, 11, 64)])
+@pytest.mark.xfail(reason="Cosine not supported in TTIR")
+def test_cosine(shapes):
+    class Cosine(nn.Module):
+        def __init__(self):
+            super().__init__()
+
+        def forward(self, a):
+            return torch.cos(a)
+
+    inputs = [torch.rand(shapes)]
+
+    framework_model = Cosine()
+    fw_out = framework_model(*inputs)
+
+    compiled_model = forge.compile(framework_model, sample_inputs=inputs)
+    co_out = compiled_model(*inputs)
+
+    co_out = [co.to("cpu") for co in co_out]
+    assert compare_with_golden_pcc(golden=fw_out, calculated=co_out[0], pcc=0.99)
+
+
+@pytest.mark.parametrize("shapes", [(1, 11, 64)])
+@pytest.mark.xfail(reason="Sine not supported in TTIR")
+def test_sine(shapes):
+    class Sine(nn.Module):
+        def __init__(self):
+            super().__init__()
+
+        def forward(self, a):
+            return torch.sin(a)
+
+    inputs = [torch.rand(shapes)]
+
+    framework_model = Sine()
+    fw_out = framework_model(*inputs)
+
+    compiled_model = forge.compile(framework_model, sample_inputs=inputs)
+    co_out = compiled_model(*inputs)
+
+    co_out = [co.to("cpu") for co in co_out]
+    assert compare_with_golden_pcc(golden=fw_out, calculated=co_out[0], pcc=0.99)
+
+
+@pytest.mark.parametrize(
+    "shapes",
+    [
+        ((1, 11), 128256, 2048),
+    ],
+)
+@pytest.mark.xfail(reason="ttnn.embedding op fails while reshaping the input_tensor in TILE_LAYOUT")
```

Comment:
Is this still the case? The latest MLIR uplift should have solved this one. Let's double-check before proceeding.

Reply:
As far as i know, it still has the same issue. Maybe i am missing something, but the test still fails